src/scripts/oneNoteApiBase.ts

```diff
@@ -48,20 +48,21 @@ export class OneNoteApiBase {
 		return apiRootUrl + partialUrl;
 	}
 
-	private makeRequest(url: string, data?: XHRData, contentType?: string): Promise<ResponsePackage<any> | OneNoteApi.RequestError> {
+	private makeRequest(url: string, data?: XHRData, contentType?: string, verb?: string): Promise<ResponsePackage<any> | OneNoteApi.RequestError> {
 		return new Promise((resolve: (responsePackage: ResponsePackage<any>) => void, reject: (error: OneNoteApi.RequestError) => void) => {
 			let request = new XMLHttpRequest();
 
-			let type: string = data ? "POST" : "GET";
+			let type: string = verb ? verb : data ? "POST" : "GET";
```

Comment:
I'm actually tempted to say we should make the verb a compulsory argument and have it explicitly defined. It feels like there's too many factors affected what the http method might end up being.